### PR TITLE
Explicitly pad start delimiters on Windows

### PR DESCRIPTION
**Pull request by ranweiler**
Sancov table start delimiters are padded on Windows, but not Linux. Expose this in the `SancovDelimiters` interface and update the PE block provider.

Based on #855, merge after.

### CODE REVIEW

File: src/agent/coverage/src/sancov.rs [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+
+    define_table_getter!(
+        name = msvc_preview_counters_table,
+        start = msvc_preview_counters_start,
+        stop = msvc_preview_counters_stop,
+        ty = SancovTableTy::Counters,
+    );
+
+    pub fn insert(&mut self, delimiter: Delimiter, offset: u32) {
+        use Delimiter::*;
```
Comment (bmc-msft) [inline src/agent/coverage/src/sancov.rs:170]: Importing from an enum defined in this file is a unusual pattern that I don't think we do anywhere else.  I recognize this is copied from a previous commit, but seeing it here I found it odd.

File: src/agent/coverage/src/sancov.rs [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+    MsvcPcsStop,
+    MsvcPreviewCountersStart,
+    MsvcPreviewCountersStop,
+}
+
+/// Size of padding inserted (on Window) between `__start_` delimiter symbols
+/// and the first entry of the delimited table's array.
+///
+/// To find the true start offset of the table, add this to the symbol value.
+const DELIMITER_START_PADDING: u32 = 8;
```
Comment (bmc-msft) [inline src/agent/coverage/src/sancov.rs:288]: Can we move this earlier in the file?  I thought you missed it in the reorg when reviewing define_table_getter

File: src/agent/coverage/src/sancov.rs [hunk 3]
```diff
@@ -67,58 +66,54 @@ macro_rules! define_table_getter {
...
         }
 
-        if let Some(table) = self.msvc_counters_table() {
+        if let Some(table) = self.msvc_counters_table(pad) {
             return Some(table);
         }
 
-        if let Some(table) = self.msvc_preview_counters_table() {
+        if let Some(table) = self.msvc_preview_counters_table(pad) {
+
```
Comment (bmc-msft) [inline src/agent/coverage/src/sancov.rs:90]:
```suggestion
```

File: src/agent/coverage/src/sancov.rs [hunk 4]
```diff
@@ -67,58 +66,54 @@ macro_rules! define_table_getter {
...
         // No MSVC tables found. Check for LLVM-emitted tables.
 
-        if let Some(table) = self.llvm_bools_table() {
+        if let Some(table) = self.llvm_bools_table(pad) {
             return Some(table);
         }
 
-        if let Some(table) = self.llvm_counters_table() {
+        if let Some(table) = self.llvm_counters_table(pad) {
+
```
Comment (bmc-msft) [inline src/agent/coverage/src/sancov.rs:101]:
```suggestion
```

File: src/agent/coverage/src/sancov.rs [hunk 5]
```diff
@@ -67,58 +66,54 @@ macro_rules! define_table_getter {
...
+    pub fn pcs_table(&self, pad: bool) -> Option<SancovTable> {
         // Check for MSVC tables first.
-        if let Some(table) = self.msvc_pcs_table() {
+        if let Some(table) = self.msvc_pcs_table(pad) {
             return Some(table);
         }
 
-        if let Some(table) = self.llvm_pcs_table() {
+        if let Some(table) = self.llvm_pcs_table(pad) {
+
```
Comment (bmc-msft) [inline src/agent/coverage/src/sancov.rs:116]:
```suggestion
```

File: src/agent/coverage/src/sancov.rs (hunk 1 above)
Comment (ranweiler) [reply src/agent/coverage/src/sancov.rs]: Changed. There are a small number of other places where we do it, but it's not super high value here.

**Review by bmc-msft — APPROVED** [review]

**Pull request merged**